gcp/gcp.go

```diff
@@ -0,0 +1,67 @@
+package gcp
+
+import (
+	"cloud.google.com/go/pubsub"
+	"google.golang.org/api/option"
+)
+
+// This should satify interface for FirehoseBackend
+type Backend struct {
+	client   *pubsub.Client
+	settings Settings
+}
+
+// SubscriptionProject represents a tuple of subscription name and project for cross-project Google subscriptions
+type SubscriptionProject struct {
+	// Subscription name
+	Subscription string
+
+	// ProjectID
+	ProjectID string
+}
+
+// Settings for Hedwig firehose
+type Settings struct {
+	// bucket where leader file is saved
+	MetadataBucket string
+
+	// bucket where follower put intermediate files to be moved by leader
+	StagingBucket string
+
+	// final bucket for firehose files
+	OutputBucket string
+
+	// interval when leader moves files to final bucket
+	ScrapeInterval int
```

Comment:
im thinking scrapeInterval and FlushAfter have nothing to do with the cloud backend  and can be moved out but wasn't exactly sure what you were thinking for those @maroux

Reply:
Yeah sounds like these should be moved out into the higher level of firehose config perhaps.

Reply:
moved to higher lvl